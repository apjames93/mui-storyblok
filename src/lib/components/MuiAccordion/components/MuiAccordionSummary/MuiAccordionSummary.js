import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { AccordionSummary } from '@material-ui/core';
import Storyblok from '../../../../utils/Storyblok';
import MuiAccordionTypography from '../MuiAccordionTypography/MuiAccordionTypography';
import MuiIcon from '../../../MuiIcon/MuiIcon';
import { validComponents } from '../../../../utils/customProps';
import { renderComponents } from '../../../../utils/customComponents';

const MuiAccordionSummary = ({
  rootClass,
  content,
  expandIcon,
}) => {
  const components = {
    MuiAccordionTypography,
    MuiIcon,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <AccordionSummary
      className={styles.root}
      expandIcon={createElement(components[expandIcon[0].component], expandIcon[0])}
    >
      {content.map((component, key) => renderComponents(components, component, key))}
    </AccordionSummary>
  );
};

export default MuiAccordionSummary;

MuiAccordionSummary.propTypes = {
  /** storyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** MuiIcon limit 1 */
  expandIcon(props, propName, componentName) {
    const components = ['MuiIcon'];
    return validComponents(props, propName, componentName, components, 1);
  },
  /** MuiAccordionTypography */
  content(props, propName, componentName) {
    const components = ['MuiAccordionTypography'];
    return validComponents(props, propName, componentName, components);
  },
};

MuiAccordionSummary.defaultProps = {
  rootClass: [],
  content: [],
  expandIcon: [],
};
