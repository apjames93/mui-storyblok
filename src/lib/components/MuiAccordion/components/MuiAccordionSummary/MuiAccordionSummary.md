##### MuiAccordionSummary has the ability to render custom components.

MuiAccordionSummary Example:

```js

const expandIcon = [{
    component: 'MuiIcon',
    iconName: 'keyboard_arrow_down',
}];

const content = [{
  component: 'MuiAccordionTypography',
  content: [{
    component: 'MuiText',
    content: 'This is MuiAccordionTypography in MuiAccordionSummary.',
  }],
}];

<MuiAccordionSummary
  content={content}
  expandIcon={expandIcon}
/>
```