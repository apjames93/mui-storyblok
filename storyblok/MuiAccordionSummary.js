module.exports = {
  "name": "MuiAccordionSummary",
  "display_name": null,
  "updated_at": "2020-07-20T17:43:17.498Z",
  "schema": {
    "rootClass": {
      "type": "options"
    },
    "expandIcon": {
      "type": "bloks",
      "max_length": "",
      "maximum": "1",
      "restrict_components": true,
      "component_whitelist": [
        "MuiIcon"
      ]
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiAccordionTypography"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiAccordionSummary",
  "component_group_uuid": null
}