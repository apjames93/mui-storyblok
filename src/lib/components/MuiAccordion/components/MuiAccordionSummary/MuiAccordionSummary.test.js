import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiAccordionSummary from './MuiAccordionSummary';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiAccordionSummary", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    component: 'MuiAccordionSummary',
    content: [{
      _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiIcon',
      iconName: 'android',
    }, {
      _editable: '<!--#storyblok#{"name": "MuiAccordionTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiAccordionTypography',
      content: [{
        _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiText',
        text: 'text',
      }],
    }],
    expandIcon: [{
      component: 'MuiIcon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<MuiAccordionSummary {...props} />);
  return { comp, props };
}

describe('<MuiAccordionSummary />', () => {
  it('renders MuiAccordionSummary', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiAccordionSummary {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
