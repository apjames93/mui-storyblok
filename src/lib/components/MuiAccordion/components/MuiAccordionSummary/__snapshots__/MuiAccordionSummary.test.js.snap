// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiAccordionSummary /> snapshot 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-root-3"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiAccordionSummary-content"
  >
    <div
      className="MuiIcon storyblok--outline"
      data-blok-c="{\\"name\\": \\"MuiIcon\\", \\"space\\": \\"48408\\", \\"uid\\": \\"947be9f0-47c3-4315-a95a-550f0c560eb5\\", \\"id\\": \\"307934\\"}"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-4 MuiIcon-colorSecondary"
      >
        android
      </span>
    </div>
    <div
      className="MuiAccordionTypography storyblok--outline"
      data-blok-c="{\\"name\\": \\"MuiAccordionTypography\\", \\"space\\": \\"48408\\", \\"uid\\": \\"947be9f0-47c3-4315-a95a-550f0c560eb5\\", \\"id\\": \\"307934\\"}"
    >
      <div
        className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
      >
        <span
          className="MuiTypography-root makeStyles-root-6 MuiTypography-body1"
        >
           undefined 
        </span>
      </div>
    </div>
  </div>
  <div
    aria-disabled={false}
    aria-hidden={true}
    className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role={null}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-7 MuiIcon-colorSecondary"
      >
        android
      </span>
    </span>
  </div>
</div>
`;
